backend/apis/events/add-event.api.js

```diff
@@ -0,0 +1,65 @@
+const mysql = require("mysql");
+const { app, databaseError, pool, invalidRequest } = require("../../server");
+const {
+  checkValid,
+  nonEmptyString,
+  validDate,
+  validInteger
+} = require("../utils/validation-utils");
+
+app.post("/api/events", (req, res, next) => {
+  pool.getConnection((err, connection) => {
+    if (err) {
+      return databaseError(req, res, err, connection);
+    }
+
+    const validityErrors = checkValid(
```

Comment:
If the request is invalid, let's not even bother getting a database connection only to immediately release it. In other words, let's move the validity check to be above the `pool.getConnection()` call